src/modules/oh-indicators-list/network-layer/sseClient.ts

```diff
@@ -1,18 +1,41 @@
+interface IEventHandlers {
+  onMessageHandler: (event: MessageEvent) => void;
+  onOpenHandler: () => void;
+  onErrorHandler: () => void;
+}
 export interface IEventSourceInitializer {
   init: (
-    eventName: string,
-    eventHandlers: {
-      onMessageHandler: (event: MessageEvent) => void;
-      onOpenHandler: (eventName: string) => void;
-      onErrorHandler: (e: any) => void;
-    },
+    url: string,
+    eventHandlers: IEventHandlers,
   ) => void;
 }
 
 export interface IEventSourceFinalizer {
-  destroy: (eventName: string) => void;
+  destroy: () => void;
 }
 
-export default class {
-    // TODO implementation
-}
\ No newline at end of file
+export const parseStreamingResponse = (
+  incomingMessage: MessageEvent,
+): { name: string; value: string } => {
+  const data = JSON.parse(incomingMessage.data);
+  const [, , name] = data.topic.split('/');
+  const { value } = JSON.parse(data.payload);
+  return { name, value };
+};
```

Comment:
Let's delegate this parsing to the reducer level? Redux guys recommend keeping the state shape responsibility of the reducer and not applying the formatting at the action level

Reply:
you can remove it from here now